lib/active_record/connection_adapters/crate_adapter.rb

```diff
@@ -131,11 +131,16 @@ def connect
 
       def columns(table_name) #:nodoc:
         cols = @connection.table_structure(table_name).map do |field|
-          CrateColumn.new(field[2], nil, field[4], nil)
+          name = dotted_name(field[2])
+          CrateColumn.new(name, nil, field[4], nil)
         end
         cols
       end
 
+      def dotted_name(name)
```

Comment:
I expected nothing less than a request for a test. :smile: Unfortunately it's more difficult than I would like it to be. The [CrateAdapter#initialize](https://github.com/crate/activerecord-crate-adapter/blob/e23cf230b46133fb39c19345229af7bb725f6fa9/lib/active_record/connection_adapters/crate_adapter.rb#L83-L91) method does a lot of stuff. I cannot create an Adapter without a real connection or too much stubing.

I don't want to refactor the Adapter to use some kind of Dependency Injection for easier testing (the adapter has currently no tests at all). My suggestion is either to leave it as is (there is an [integration test](https://github.com/crate/activerecord-crate-adapter/blob/e23cf230b46133fb39c19345229af7bb725f6fa9/spec/data_types/object_spec.rb#L48) which executes this method) or I could extract the method into some kind of Helper/Util-Class where it's trivial to test. You decide.